model/task_lifecycle.go

```diff
@@ -905,6 +905,11 @@ func MarkEnd(ctx context.Context, settings *evergreen.Settings, t *task.Task, ca
 		}))
 	}
 
+	// Deactivate previous occurrences of the same task if this one passed on mainline commits.
+	if t.Status == evergreen.TaskSucceeded && deactivatePrevious && t.Requester == evergreen.RepotrackerVersionRequester {
```

Comment:
This will also need to check if the task is not part of an existing stepback process, i.e. that the activator is not the stepback one.